src/ra_server_proc.erl

```diff
@@ -1872,11 +1874,20 @@ handle_process_down(Pid, Info, RaftState,
                                     monitors = Monitors}),
     {keep_state, State, Actions}.
 
-record_leader_change(Leader, #state{conf = #conf{cluster_name = ClusterName},
-                                    server_state = ServerState}) ->
-    Members = do_state_query(members, ServerState),
-    ok = ra_leaderboard:record(ClusterName, Leader, Members),
-    ok.
+record_cluster_change(
+  #state{conf = #conf{cluster_name = ClusterName},
+         server_state = ServerStateA},
+  #state{server_state = ServerStateB}) ->
+    LeaderA = do_state_query(leader, ServerStateA),
+    MembersA = do_state_query(members, ServerStateA),
```

Comment:
It should be sufficient to compare the map size of the before and after clusters which should be more efficient than a query and equality check. 

This is because we can only change one member at a time.

Reply:
I think this is addressed by your changes.